From 2301ad66f0afc2da6f8cb21ce8d9a06b27f9a93b Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Thu, 17 Aug 2023 10:59:53 +0200
Subject: libstdc++: fpclassify exposed by cmath is in std namespace
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix build with libstdc++, as in it, fpclassify is not exposed directly
but in the std namespace.

Fixes this build error with libstdc++:
    ../../base/allocator/partition_allocator/partition_alloc_base/strings/cstring_builder.cc: In member function ‘void partition_alloc::internal::base::strings::CStringBuilder::PutFloatingPoint(double, unsigned int)’:
    ../../base/allocator/partition_allocator/partition_alloc_base/strings/cstring_builder.cc:150:11: error: ‘fpclassify’ was not declared in this scope; did you mean ‘std::fpclassify’?
      150 |   switch (fpclassify(value)) {
          |           ^~~~~~~~~~
          |           std::fpclassify

Bug: 957519
Change-Id: Ia38b68cf1b5c04bb5169db1c669d32ff6fc48525
---
 .../partition_alloc_base/strings/cstring_builder.cc             | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/base/allocator/partition_allocator/partition_alloc_base/strings/cstring_builder.cc b/base/allocator/partition_allocator/partition_alloc_base/strings/cstring_builder.cc
index 91fa0e0cfbfc8..2643cda4c1e3e 100644
--- a/base/allocator/partition_allocator/partition_alloc_base/strings/cstring_builder.cc
+++ b/base/allocator/partition_allocator/partition_alloc_base/strings/cstring_builder.cc
@@ -147,7 +147,7 @@ const char* CStringBuilder::c_str() {
 }
 
 void CStringBuilder::PutFloatingPoint(double value, unsigned num_digits10) {
-  switch (fpclassify(value)) {
+  switch (std::fpclassify(value)) {
     case FP_INFINITE:
       PutText(value < 0 ? "-inf" : "inf");
       break;
-- 
2.39.2

